shared/profile/edit-profile/index.js

```diff
@@ -1,65 +1,42 @@
 // @flow
-import React, {Component} from 'react'
 import Render from './render'
+import {compose, withHandlers, withPropsOnChange, withState} from 'recompose'
 import {connect} from 'react-redux'
 import {editProfile} from '../../actions/profile'
 import {maxProfileBioChars} from '../../constants/profile'
 import {navigateUp} from '../../actions/route-tree'
 
-import type {Props} from './render'
-
-class EditProfile extends Component<void, Props, void> {
-  render() {
-    const bioMaxChars = maxProfileBioChars
-    const bioLengthLeft = bioMaxChars - this.props.bio.length
-    return (
-      <Render
-        bio={this.props.bio}
-        bioLengthLeft={bioLengthLeft}
-        fullname={this.props.fullname}
-        location={this.props.location}
-        onBack={this.props.onBack}
-        onBioChange={this.props.onBioChange}
-        onCancel={this.props.onBack}
-        onFullnameChange={this.props.onFullnameChange}
-        onLocationChange={this.props.onLocationChange}
-        onSubmit={this.props.onEditProfile}
-      />
-    )
-  }
-}
+const RenderWrapped = compose(
```

Comment:
k 3 more changes will make this very clean

1. can you move the mapStateToProps and mapDispatchToProps to their own helpers, then you can do
connect(mapStateToProps, mapDispatchToProps)
1. You don't need the merge function as you can just change the dispatch onEditProfile take the params like the merge one does
1. you can use compose w/ connect, so you can do 
```
compose(
connect(mapStateToProps, mapDispatchToProps)
withState('bio',...,
...
)
```